auditbeat/module/file_integrity/schema.fbs

```diff
@@ -61,13 +61,49 @@ table Hash {
   xxh64: [byte];
 }
 
+table Section {
+  name:string;
+  physical_size:ulong;
+  virtual_size:ulong;
+  entropy:float64;
+  var_entropy:float64;
+}
+
+table ExecutableObject {
+  // Import hashes
+  import_hash:[byte];
+  imphash:[byte];
+  symhash:[byte];
+  go_import_hash:[byte];
+
+  // Name entropies
+  import_names_entropy:float64;
+  go_import_names_entropy:float64;
+
+  // Import list
+  imports:[string];
+  go_imports:[string];
+
+  // Section statistics
+  sections:[Section];
+
+  go_stripped:bool;
+}
+
+table ParserResult {
+  pe:ExecutableObject;
+  elf:ExecutableObject;
+  macho:ExecutableObject;
+}
+
 table Event {
   timestamp_ns:long;
   target_path:string;
   action:Action;
   source:Source;
   info:Metadata;
   hashes:Hash;
+  file:ParserResult;
```

Comment:
I think I may have raised this before, but it's been a while so please forgive me if I did.

Do we need to persist the file parser results? Persisted data is used in determining whether or not to generate an event (comparison in `diffEvents()`). Wouldn't a change in the overall file hash be sufficient to determine if we to generate an event?

I'm not proposing that you make this change here: Taking this a step further, we could even defer the file parsing work until we know there was a hash change.

Reply:
Yes, that makes sense. I'd say that we don't need to persist it and we should only do the work if the file hash has changed.